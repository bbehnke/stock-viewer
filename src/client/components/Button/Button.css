.stock-viewer-button {
  background: white;
  color: #9068be;
  border: 1px #9068be solid;
  padding: 5px 10px;
  border-radius: 2px;
  font-weight: bold;
  font-size: 9pt;
  outline: none;
  cursor: pointer;
  height: 30px;
}

.stock-viewer-button:hover {
  background: #9068be;
  color: white;
}

.stock-viewer-button:focus {
  box-shadow: 1px 1px 1px #9068be;
}

.stock-viewer-button:disabled {
  opacity: 0.65;
  cursor: not-allowed;
}

.stock-viewer-button:disabled:hover {
  background: white;
  color: #9068be;
}
