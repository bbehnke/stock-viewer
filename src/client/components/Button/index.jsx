import React from 'react';
import PropTypes from 'prop-types';
import './Button.css';

const Button = ({
  value, onClick, className, disabled
}) => (
  <button
    className={`stock-viewer-button ${className}`}
    type="button"
    onClick={onClick}
    disabled={disabled}
  >
    {value}
  </button>
);

Button.defaultProps = {
  onClick: () => {},
  className: '',
  disabled: false
};

Button.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element
  ]).isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
  disabled: PropTypes.bool
};

export default Button;
