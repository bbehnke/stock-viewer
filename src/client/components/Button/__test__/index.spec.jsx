import React from 'react';
import { mount } from 'enzyme';
import Button from '..';

describe('Button tests', () => {
  it('renders Button', () => {
    const component = mount(<Button className="test-class" value="test-value" />);
    expect(component).toMatchSnapshot();
  });
  it('renders Button disabled', () => {
    const component = mount(<Button disabled className="test-class" value="test-value" />);
    expect(component).toMatchSnapshot();
  });
  it('onClick called', () => {
    const onClickMock = jest.fn();
    const component = mount(
      <Button
        className="test-class"
        value="test-value"
        onClick={onClickMock}
      />
    );
    component.find('button.stock-viewer-button').simulate('click');
    // component.find('button.note-editor-submit').simulate('submit');
    expect(onClickMock).toBeCalledTimes(1);
  });
});
