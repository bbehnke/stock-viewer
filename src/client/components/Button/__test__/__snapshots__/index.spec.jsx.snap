// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders Button 1`] = `
<Button
  className="test-class"
  disabled={false}
  onClick={[Function]}
  value="test-value"
>
  <button
    className="stock-viewer-button test-class"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    test-value
  </button>
</Button>
`;

exports[`Button tests renders Button disabled 1`] = `
<Button
  className="test-class"
  disabled={true}
  onClick={[Function]}
  value="test-value"
>
  <button
    className="stock-viewer-button test-class"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    test-value
  </button>
</Button>
`;
